### PR TITLE
Adding support for Framework 4.6.1 SDK.

### LINKED ISSUE #177: Add support for .NET 4.61 Framework SDK
The change needs to ship in Dev 14 Update 1

New Path for registry: `HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\Microsoft SDKs\NETFXSDK\4.6.1`
New Path for Tools: `C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.6.1 Tools`
New Path for Kits: `C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1`

We also want to include an overload for `ToolLocationHelper.GetPathToDotNetFrameworkSdkFile` to always use the latest (currently it takes an enum and Latest will only update if you recompile both sides).

### PR DESCRIPTION
**Pull request by AndyGerlicher**
This closes #177.

### CODE REVIEW

File: src/XMakeTasks/GetFrameworkSDKPath.cs [hunk 1]
```diff
@@ -269,6 +270,40 @@ public string FrameworkSdkVersion46Path
             }
         }
 
+        /// <summary>
+        /// The path to the v4.6 .NET SDK if it could be found. It will be String.Empty if the SDK was not found.
```
Comment (ValMenn) [inline src/XMakeTasks/GetFrameworkSDKPath.cs]: Comment should say v4.61

File: src/Shared/FrameworkLocationHelper.cs [hunk 2]
```diff
@@ -1126,6 +1123,25 @@ public string WindowsSdkRegistryInstallationFolderName
             {
                 get { return _windowsSdkRegistryInstallationFolderName; }
             }
+
+            /// <summary>
+            /// The key in registry to indicate the corresponding .net framework in this visual studio.
```
Comment (rainersigwald) [inline src/Shared/FrameworkLocationHelper.cs]: in *the* registry

Since we're changing it anyway :)

File: src/Shared/FrameworkLocationHelper.cs [hunk 3]
```diff
@@ -1126,6 +1123,25 @@ public string WindowsSdkRegistryInstallationFolderName
...
+
+                // Framework 4.6.1 
+                if (dotNetSdkVersion == dotNetFrameworkVersion461)
+                {
+                    sdkVersionFolder = "4.6.1";
+                }
+
+                // If the path is formatted to include a version number, we need to include that.
+                // (e.g. NETFXSDK\{0} should be NETFXSDK\4.6 or NETFXSDK\4.6.1
+                return string.Format(_dotNetFrameworkSdkRegistryKey, sdkVersionFolder);
```
Comment (rainersigwald) [inline src/Shared/FrameworkLocationHelper.cs]: Comment that most strings don't have the format specifier and come out unmodified.

**Comment by rainersigwald** [comment]
LGTM

**Pull request merged**